src/main/app/controller/courts/AddressController.ts

```diff
@@ -0,0 +1,276 @@
+      await this.render(req, res, false, addresses, [this.updateAddressError]);
+      return;
+    }
+
+    // Validate addresses
+    const primaryValidationResult = this.validateAddress(addresses.primary, true);
+    const secondaryValidationResult = this.validateAddress(addresses.secondary, false);
+    const addressTypeErrors  = this.validateAddressTypes(addresses.primary, addresses.secondary, writeToUsTypeId);
+    const addressesValid = primaryValidationResult.addressValid && primaryValidationResult.postcodeValid &&
+      secondaryValidationResult.addressValid && secondaryValidationResult.postcodeValid && addressTypeErrors.length === 0;
+
+    if (!addressesValid) {
+      const allErrors = primaryValidationResult.errors.concat(secondaryValidationResult.errors).concat(addressTypeErrors);
+      await this.render(req, res, false, addresses, allErrors,
+        !primaryValidationResult.postcodeValid, !secondaryValidationResult.postcodeValid);
+      return;
+    }
+
+    // Post addresses to API if valid
+    await req.scope.cradle.api.updateCourtAddresses(req.params.slug, this.convertToApiType(addresses))
+      .then(async (addressList: CourtAddress[]) => await this.render(req, res, true, this.convertToDisplayAddresses(addressList)) )
+      .catch(async (reason: AxiosError) => {
+        if (reason.response.status === 400) {
+          const postcodeValidation = this.checkErrorResponseForPostcodeErrors(reason, addresses);
+          const errors = postcodeValidation.errors.length === 0
+            ? [this.updateAddressError] // we've encountered a 400 for a reason other than postcodes
+            : postcodeValidation.errors;
+          await this.render(req, res, false, addresses, errors, postcodeValidation.primaryInvalid, postcodeValidation.secondaryInvalid);
+        } else {
+          await this.render(req, res, false, addresses, [this.updateAddressError]);
+        }
+      });
+  }
+
+  private async render(
+    req: AuthedRequest,
+    res: Response,
+    updated = false,
+    addresses: DisplayCourtAddresses = null,
+    errorMsgs: string[] = [],
+    primaryPostcodeInvalid = false,
+    secondaryPostcodeInvalid = false) {
+
+    const slug: string = req.params.slug as string;
+    let fatalError = false;
+
+    if (!addresses) {
+      await req.scope.cradle.api.getCourtAddresses(slug)
+        .then((addressList: CourtAddress[]) => addresses = this.convertToDisplayAddresses(addressList))
+        .catch(() => {
+          errorMsgs.push(this.getAddressesError);
+          fatalError = true;
+        });
+    }
+
+    let addressTypes: AddressType[] = [];
+    await req.scope.cradle.api.getAddressTypes()
+      .then((types: AddressType[]) => addressTypes = types)
+      .catch(() => {
+        errorMsgs.push(this.getAddressTypesError);
+        fatalError = true;
+      });
+    const writeToUsTypes = addressTypes.filter(at => at.name.toLowerCase() === this.writeToUsAddressType.toLowerCase()).map(at => at.id);
```

Comment:
Can you add a comment here to describe the why behind the importance of this particular address type. It may also be useful to note why we are only interested in the first index. 

If this comes back as more than one element in the array, do we need to consider the other values?

Reply:
Yeah - I agree some explanation required. Updated with comment.